datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -63,3 +62,21 @@ def normalize_datetime(dt):
         dt = dt.replace(tzinfo=pytz.utc)
 
     return dt
+
+
+class ConstantRateLimiter:
+    def __init__(self, rate_limit_s):
+        """
+        :param rate_limit_s: rate limit in seconds
+        """
+        self.period_s = 1 / rate_limit_s if rate_limit_s > 0 else 0
+        self.last_event = 0
+
+    def sleep(self):
```

Comment:
Conventionally, these sorts of data structures block on consuming a resource, not blocking after the action is taken. This is more or less fine for the current use in plans, but a sleep method isn't really generally useful.

Reply:
yeah makes sense, moved to the top